app/jobs/maintenance_tasks/task_job.rb

```diff
@@ -97,8 +97,18 @@ def on_shutdown
       @ticker.persist
     end
 
+    def reenqueue_iteration_job
+      if defined?(@reenqueue_iteration_job) && @reenqueue_iteration_job
+        super
+      end
+    end
+
     def after_perform
       @run.save!
+      if @run.interrupted?
```

Comment:
Instead of relying on the state of the run, maybe we should just have to code only focused on delaying the enqueuing? i.e. instead of checking for the situation where the job would have be re-enqueued (`@run.interrupted?`), remember if re-enqueuing was tried when it's prevented and then later check whether it would have been re-enqueued and only re-enqueuing if it was.

Reply:
Sounds good, I was torn between introducing a new ivar to handle this vs using something we already had (the run status), but if we upstream later it might be better to shift this logic to be focused on the delayed enqueue instead 👍

Reply:
Oh we don't even need another variable to do this, we can just set `@reenqueue_iteration_job` in `reenqueue_iteration_job` if it's not yet set, duh 🤦‍♀️